compiler/parser/RdrHsSyn.hs

```diff
@@ -3729,6 +3737,30 @@ mkInterfaceInstanceDecls parentName sharedBinds (L loc interfaceInstance) = do
         , noLoc (ValD noExt val)
         ]
 
+renameMatchGroup ::
+     Located RdrName
+  -> MatchGroup GhcPs (LHsExpr GhcPs)
+  -> MatchGroup GhcPs (LHsExpr GhcPs)
+renameMatchGroup newName = \case
+  MG { mg_alts, .. } -> MG
+    { mg_alts = fmap (fmap (fmap renameMatch)) mg_alts
+    , ..
+    }
+  x -> x
+  where
+    renameMatch = \case
+      Match { m_ctxt, .. } -> Match
+        { m_ctxt = renameMatchContext m_ctxt
+        , ..
+        }
+      x -> x
+    renameMatchContext = \case
+      FunRhs { .. } -> FunRhs
+        { mc_fun = newName
+        , ..
+        }
+      x -> x
```

Comment:
I don't know if there's a better way to do this 🤷

Reply:
looks alright to me - do you want to inline the two pattern matching functions into one case statement, or does that turn out ugly due to nested record updates?

Reply:
hm I tried that now and yes it gets a bit ugly. I was hoping there was a `Data.Data`-esque function or something, but plain pattern matching is probably better anyway.